tests/conftest.py

```diff
@@ -357,33 +357,23 @@ def _fail_for_invalid_template_variable_improved(_fail_for_invalid_template_vari
 @pytest.fixture(autouse=True, scope="function")
 def unknown_variable_template_error(monkeypatch, request):
     marker = request.keywords.get("ignore_unknown_variable_template_error", None)
-    # debug can be injected by django.template.context_processors.debug
-    # user can be injected by django.contrib.auth.context_processors.auth
-    # TODO(xfernandez): remove user from allow list (and remove the matching processor ?)
-    BASE_IGNORE_LIST = {"debug", "user"}
-    strict = True
-    if marker is None:
-        ignore_list = BASE_IGNORE_LIST
-    elif marker.args:
-        ignore_list = BASE_IGNORE_LIST | set(marker.args)
-    else:
-        # Marker without list
-        strict = False
-
-    if strict:
-        origin_resolve = base_template.FilterExpression.resolve
-
-        def stricter_resolve(self, context, ignore_failures=False):
-            if (
-                self.is_var
-                and self.var.lookups is not None
-                and self.var.lookups[0] not in context
-                and self.var.lookups[0] not in ignore_list
-            ):
-                ignore_failures = False
-            return origin_resolve(self, context, ignore_failures)
-
-        monkeypatch.setattr(base_template.FilterExpression, "resolve", stricter_resolve)
+    ignore_list = set()
+    if marker and marker.args:
+        ignore_list.update(marker.args)
+
+    origin_resolve = base_template.FilterExpression.resolve
+
+    def stricter_resolve(self, context, ignore_failures=False):
+        if (
+            self.is_var
+            and self.var.lookups is not None
+            and self.var.lookups[0] not in context
+            and self.var.lookups[0] not in ignore_list
+        ):
+            ignore_failures = False
+        return origin_resolve(self, context, ignore_failures)
+
+    monkeypatch.setattr(base_template.FilterExpression, "resolve", stricter_resolve)
```

Comment:
Proposition pour échouer quand on ignore pour rien, ça t'évitera de te taper le nettoyage à chaque fois ;).
```suggestion
    origin_resolve = base_template.FilterExpression.resolve

    seen_variables = set()

    def stricter_resolve(self, context, ignore_failures=False):
        if self.is_var and self.var.lookups is not None:
            seen_variables.add(self.var.lookups[0])
            if self.var.lookups[0] not in context and self.var.lookups[0] not in ignore_list:
                ignore_failures = False
        return origin_resolve(self, context, ignore_failures)

    monkeypatch.setattr(base_template.FilterExpression, "resolve", stricter_resolve)
    yield

    ignored_variable_not_seen = ignore_list - seen_variables
    if ignored_variable_not_seen:
        pytest.fail(f"Unnecessary usage of 'ignore_unknown_variable_template_error' for: {ignored_variable_not_seen}")
```